node/core/dispute-coordinator/src/real/participation/queues/mod.rs

```diff
@@ -206,3 +247,115 @@ impl Queues {
 		}
 	}
 }
+
+/// Entry for the best effort queue.
+struct BestEffortEntry {
+	req: ParticipationRequest,
+	/// How often was the above request added to the queue.
+	added_count: BestEffortCount,
+}
+
+/// `Comparator` for ordering of disputes for candidates.
+///
+/// This `comparator` makes it possible to order disputes based on age and to ensure some fairness
+/// between chains in case of equally old disputes.
+///
+/// Objective ordering between nodes is important in case of lots disputes, so nodes will pull in
+/// the same direction and work on resolving the same disputes first. This ensures that we will
+/// conclude some disputes, even if there are lots of them. While any objective ordering would
+/// suffice for this goal, ordering by age ensures we are not only resolving disputes, but also
+/// resolve the oldest one first, which are also the most urgent and important ones to resolve.
+///
+/// Note: That by `oldest` we mean oldest in terms of relay chain block number, for any block
+/// number that has not yet been finalized. If a block has been finalized already it should be
+/// treated as low priority when it comes to disputes, as even in the case of a negative outcome,
+/// we are already too late. The ordering mechanism here serves to prevent this from happening in
+/// the first place.
+#[derive(Copy, Clone)]
+#[cfg_attr(test, derive(Debug))]
+struct CandidateComparator {
```

Comment:
This got moved here, because scraper no longer cares about the actual comparator. All it does, is provide info on whether a candidate has been included anywhere.